packages/wallet-sdk/src/core/communicator/util.test.ts

```diff
@@ -0,0 +1,91 @@
+import { closePopup, openPopup } from './util';
+
+describe('Communicator util', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('openPopup', () => {
+    it('opens a popup window', () => {
+      const url = new URL('https://keys.coinbase.com');
+      const mockFocus = jest.fn();
+
+      const windowSpy = jest.spyOn(window, 'open').mockImplementation(() => {
+        return {
+          focus: mockFocus,
+        } as any;
+      });
+
+      openPopup(url);
+
+      expect(windowSpy).toHaveBeenCalledTimes(1);
+      expect(windowSpy).toHaveBeenCalledWith(url, 'Smart Wallet', expect.any(String));
+      expect(mockFocus).toHaveBeenCalled();
+    });
+
+    it('uses the correct dimensions and location for the popup window', () => {
+      const url = new URL('https://keys.coinbase.com');
+
+      const mockScreenXY = 100;
+      const mockScreenInnerWidthHeight = 50;
+      Object.defineProperty(window, 'screenX', { value: mockScreenXY });
+      Object.defineProperty(window, 'screenY', { value: mockScreenXY });
+      Object.defineProperty(window, 'innerWidth', { value: mockScreenInnerWidthHeight });
+      Object.defineProperty(window, 'innerHeight', { value: mockScreenInnerWidthHeight });
+
+      const windowSpy = jest.spyOn(window, 'open').mockImplementation(() => {
+        return {
+          focus: jest.fn(),
+        } as any;
+      });
+
+      openPopup(url);
+
+      const expectedWidth = 420;
+      const expectedHeight = 540;
+      const expectedLeft = (mockScreenInnerWidthHeight - expectedWidth) / 2 + mockScreenXY;
+      const expectedTop = (mockScreenInnerWidthHeight - expectedHeight) / 2 + mockScreenXY;
+      expect(windowSpy).toHaveBeenCalledWith(
+        url,
+        'Smart Wallet',
+        `width=${expectedWidth}, height=${expectedHeight}, left=${expectedLeft}, top=${expectedTop}`
+      );
+    });
```

Comment:
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/53091edf-3c37-4e3f-8d4e-65a73d6b3818)

Reply:
The test ensures that the top and left values are calculated correctly. If the calculation were to change, this test would fail and help us spot a bug

Reply:
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/9c1636df-3b69-4602-a0a7-228d7ca61fbf)
@arjun-dureja